ChemistryLib/PhreeqcKernel.cpp

```diff
@@ -0,0 +1,257 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <cmath>
+#include <cstdlib>
+
+#include "BaseLib/Error.h"
+#include "PhreeqcKernel.h"
+#include "PhreeqcKernelData/AqueousSolution.h"
+#include "PhreeqcKernelData/ReactionRate.h"
+
+#include "ThirdParty/iphreeqc/src/src/phreeqcpp/cxxKinetics.h"
+
+namespace ChemistryLib
+{
+namespace PhreeqcKernelData
+{
+PhreeqcKernel::PhreeqcKernel(
+    std::size_t const num_chemical_systems,
+    std::vector<std::pair<int, std::string>> const&
+        process_id_to_component_name_map,
+    std::string const& database,
+    PhreeqcKernelData::AqueousSolution& aqueous_solution,
+    std::unique_ptr<Kinetics>
+        kinetic_reactants,
+    std::vector<ReactionRate>&& reaction_rates)
+    : Phreeqc(),
+      _initial_aqueous_solution(aqueous_solution.getInitialAqueousSolution()),
+      _reaction_rates(std::move(reaction_rates))
+{
+    do_initialize();
```

Comment:
drop the `do_` part, it's a verb already. Would be better, instead, to add something to the name like in `reinitializeRates`, which is a good name.

Reply:
Fixed. Encapsulated into a function.